.github/workflows/test-spyre.yml

```diff
@@ -1,29 +1,133 @@
 name: test-sypre
 
-on: pull_request
+on:
+  workflow_dispatch:
+  push:
+    paths-ignore:
+      - "**.md"
+  pull_request:
+    paths-ignore:
+      - "**.md"
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  TEST_IMAGE_NAME: "vllm-spyre"
 
 jobs:
-  test-spyre:
+  build:
     runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+    env:
+      CACHE_IMAGE: "ghcr.io/vllm-project/vllm-spyre:test-cache"
+      CACHE_REGISTRY: "ghcr.io"
+
     steps:
-    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
-    - name: Build docker image
-      run: docker build . -t vllm-spyre -f Dockerfile.spyre
-    - name: Run Spyre tests within docker container
+      - name: "Checkout"
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: "Setup Docker Buildx"
+        uses: docker/setup-buildx-action@v3
+
+      - name: "Log in to cache image container registry"
+        uses: docker/login-action@v3
+        if: github.event_name != 'pull_request'
+        with:
+          registry: ${{ env.CACHE_REGISTRY }}
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: "Set build cache target"
+        run: |
+          # For push to `main` (PR merged), push new cache image with all layers (cache-mode=max).
+          # For PR builds, use GitHub action cache which isolates cached layers by PR/branch
+          # to optimize builds for subsequent pushes to the same PR/branch.
+          # Do not set a cache-to image for PR builds to not overwrite the `main` cache image
+          # to not ping-pong cache images for two or more concurrent PRs.
+          # Do not push cache images for multiple branches to not exceed GitHub package
+          # usage and traffic limitations.
+          # NOTE 2024/03/24: GHA cache appears to have issues, cannot use `cache-to: gha,mode=min`
```

Comment:
@ckadner can we re-evaluate these limitations for this image?

Based on the dates these comments would have been from our testing with the old TGIS image that we could barely fit on a GHA runner. I think this one should be much smaller, it doesn't include any of the heavyweight gpu libraries.

Reply:
ahhh, gotcha!

That's no fun :/